### PR TITLE
Remove unused DPT fusion residual parameters

### LINKED ISSUE #28292: DPT normalization causes contouring when there are significant disparities in depth values between adjacent areas
### System Info

Python 3.10.12
transformers-4.36.2

### Who can help?

@stevhliu @NielsRogge

### Information

- [X] The official example scripts
- [x] My own modified scripts

### Tasks

- [x] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [x] My own task or dataset (give details below)

### Reproduction

```
from transformers import DPTImageProcessor, DPTForDepthEstimation
import torch
import numpy as np
from PIL import Image
import requests

url = "https://images.unsplash.com/photo-1605146768851-eda79da39897?q=80&w=2970&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D"
image = Image.open(requests.get(url, stream=True).raw)

processor = DPTImageProcessor.from_pretrained("Intel/dpt-large")
model = DPTForDepthEstimation.from_pretrained("Intel/dpt-large")

# prepare image for the model
inputs = processor(images=image, return_tensors="pt")

with torch.no_grad():
    outputs = model(**inputs)
    predicted_depth = outputs.predicted_depth

# interpolate to original size
prediction = torch.nn.functional.interpolate(
    predicted_depth.unsqueeze(1),
    size=image.size[::-1],
    mode="bicubic",
    align_corners=False,
)

# visualize the prediction
output = prediction.squeeze().cpu().numpy()
formatted = (output * 255 / np.max(output)).astype("uint8")
depth = Image.fromarray(formatted)
display(depth)
```

>
Some weights of DPTForDepthEstimation were not initialized from the model checkpoint at Intel/dpt-large and are newly initialized: ['neck.fusion_stage.layers.0.residual_layer1.convolution2.bias', 'neck.fusion_stage.layers.0.residual_layer1.convolution1.weight', 'neck.fusion_stage.layers.0.residual_layer1.convolution2.weight', 'neck.fusion_stage.layers.0.residual_layer1.convolution1.bias']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.

edge_effects_depth (1)

### Expected behavior

Anecdotally, the local scaling methodology used by get_depth_map at https://huggingface.co/diffusers/controlnet-depth-sdxl-1.0 seems to work better for models that perform better at identifying close-range depth. The global scaling methodology seems to work better for models that perform better at identifying far-range depth. I combined them below:

```
def get_depth_map(image, feature_extractor, depth_estimator, scale_local):
    inputs = feature_extractor(images=image, return_tensors="pt").pixel_values.to("cuda")
    with torch.no_grad(), torch.autocast("cuda"):
        depth_map = depth_estimator(inputs).predicted_depth

    depth_map = torch.nn.functional.interpolate(
        depth_map.unsqueeze(1),
        size=image.size[::-1],
        mode="bicubic",
        align_corners=False,
    )

    if scale_local:
        depth_min = torch.amin(depth_map, dim=[1, 2, 3], keepdim=True)
        depth_max = torch.amax(depth_map, dim=[1, 2, 3], keepdim=True)
        depth_map = (depth_map - depth_min) / (depth_max - depth_min)
        image = torch.cat([depth_map] * 3, dim=1)
    
        image = image.permute(0, 2, 3, 1).cpu().numpy()[0]
        image = Image.fromarray((image * 255.0).clip(0, 255).astype(np.uint8))
        return image
    
    output = depth_map.squeeze().cpu().numpy()
    formatted = (output * 255 / np.max(output)).astype("uint8")
    return Image.fromarray(formatted)


depth_estimator_hybrid = DPTForDepthEstimation.from_pretrained("Intel/dpt-hybrid-midas").to("cuda")
depth_estimator_dinov2_nyu = DPTForDepthEstimation.from_pretrained("facebook/dpt-dinov2-giant-nyu").to("cuda")
    
image_processor_hybrid = AutoImageProcessor.from_pretrained("Intel/dpt-hybrid-midas")
image_processor_dinov2_nyu = AutoImageProcessor.from_pretrained("facebook/dpt-dinov2-giant-nyu")

# Close range depth
bad_close_result = get_depth_map(image, image_processor_hybrid, depth_estimator_hybrid, False)
good_close_result = get_depth_map(image, image_processor_hybrid, depth_estimator_hybrid, True)

# Far range depth
downscaled_image = image.resize((1024, 1024))  # This image is too big for my GPU to processdpt-dinov2-giant-nyu so I downscaled it
good_far_result = get_depth_map(downscaled_image, image_processor_dinov2_nyu, depth_estimator_dinov2_nyu, False)
bad_far_result = get_depth_map(downscaled_image, image_processor_dinov2_nyu, depth_estimator_dinov2_nyu, True)
```

`display(bad_close_result)`
globally_scaled_depth_close

`display(good_close_result)`
locally_scaled_depth_close

`display(good_far_result)`
globally_scaled_depth_far
`display(bad_far_result)`
locally_scaled_depth_far

Sufficiently blurring the image prior to detecting depth also gets rid of this, ie:

```
blurred_image = image.filter(ImageFilter.GaussianBlur(radius=5))
display(get_depth_map(blurred_image, image_processor_hybrid, depth_estimator_hybrid, False))
```

blurred_depth

### PR DESCRIPTION
**Pull request by VictorMaxWang**
[CI](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=47699)

# What does this PR do?

The first DPT feature-fusion layer is always called without a residual input, but it still registers a `residual_layer1` module. Those parameters are therefore unreachable. Checkpoints that follow the original DPT layout omit them, which produces a misleading warning that the weights were newly initialized.

This PR:

- removes the unused first residual module from the DPT depth-estimation and semantic-segmentation heads after `post_init()`, preserving the initialization order and RNG state of every retained parameter;
- ignores the corresponding legacy checkpoint keys during `from_pretrained` loading;
- updates the DPT conversion scripts so converted state dictionaries match the corrected model structure;
- adds regression coverage for both task heads and for legacy checkpoints with and without fusion BatchNorm.

The retained parameters and outputs of both task heads are bit-for-bit identical before and after this change. This addresses the maintainer-identified random-initialization warning in #28292; it does not change depth-map normalization or post-processing behavior.

Fixes #28292

## AI-assisted contribution disclosure

I used Codex to assist with repository exploration, implementation, and local test execution. I reviewed every changed line and validated the behavior locally before submission.

- Coordination and maintainer diagnosis: #28292, which is labeled `Good Second Issue`
- Existing work: I found no open pull request addressing the unused first DPT fusion residual parameters before publishing this branch
- Scope control: copied DPT fusion implementations in ZoeDepth and DepthAnything are intentionally unchanged to avoid altering their checkpoint schemas

## Validation

```text
python -m pytest -q tests/models/dpt/test_modeling_dpt.py
102 passed, 122 skipped, 2145 subtests passed
```

Additional checks:

- targeted legacy-loading coverage: 2 tests passed, 6 subtests passed
- Ruff lint and format checks passed for all 7 changed files
- copied-code consistency and import ordering checks passed
- changed-only modeling-structure check passed
- synthetic converter checks covered both the 4-key non-BatchNorm and 12-key BatchNorm legacy subtrees

Large real-world checkpoint conversion scripts were not run end to end because they require downloading their source checkpoints; their key mapping and strict-loading behavior were validated with synthetic state dictionaries.

## Code Agent Policy

Not applicable to the first-time-contributor checkbox: I am not a first-time Transformers contributor.

## Before submitting

- [ ] This PR fixes a typo or improves the docs
- [x] I read the contributor guidelines and PR checks
- [x] This change follows the diagnosis and invitation to contribute in #28292
- [x] No documentation update is required because this does not change a public API
- [x] I added and ran the necessary tests

### CODE REVIEW

**Comment by Rocketknight1** [comment]
No code agent PRs from first time contributors please!
